packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -266,6 +274,11 @@ export async function run(context: $TSContext, resourceDefinition: $TSObject) {
 
         try {
           await updateCloudFormationNestedStack(context, nestedStack, resourcesToBeCreated, resourcesToBeUpdated);
+          if (FeatureFlags.getBoolean('Overrides.project')) {
+            storeRootStackTemplate(context, nestedStack);
+            // if the only root stack updates, function is called with empty resources . this fn copies amplifyMeta and backend Config to #current-cloud-backend
+            context.amplify.updateamplifyMetaAfterPush([]);
```

Comment:
why do we need this?

Reply:
if there is only root stack override , then copying backend resources to current cloud backend, if resources are there then it will get copied again